### PR TITLE
Feature/Authentication-Posts

### PR DESCRIPTION
**Pull request by cliftondavies**
# Project: Authentication
## Milestones:
- [x] Gitflow
- [x] Github Actions
- [x] Rubocop
- [x] Stylelint
- [x] .gitignore
- [x] Sign Up feature for users to create an account
- [x] Sign-in and Sign-out links in application layout
- [x] README

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!--<p id="notice"><%= notice %></p>-->
```
Comment (maelfosso) [inline app/views/posts/index.html.erb:1]: Comments are not allowed, please.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -10,6 +10,23 @@
...
+      <%= link_to 'MembersOnly', root_path %>
+
+      <ul>
+        <% if user_signed_in? %>
+        <li>Welcome, <%= current_user.name %></li>
+        <li><%= link_to 'Sign Out', destroy_user_session_path, method: :delete %></li>
+        <% else %>
+        <li><%= link_to 'Sign In', new_user_session_path %></li>
+        <li><%= link_to 'Sign Up', new_user_registration_path %></li>
+        <% end %>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb:26]: Implement this logic into **view helpers**

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:2]: `show, edit, update, destroy` are not all allowed. Don't forget that only 3 of the HTTP method are allowed for this controller
That means some of those methods are not welcomed there.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  def create
+    @post = current_user.posts.build(post_params)
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to root_path, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
+      else
+        format.html { render :new }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:22]: Remove `format.json`. You don't need it!

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<h1>Posts</h1>
+
+<section>
+  <% @posts.each do |post| %>
+  <article>
+    <h2><%= post.title %></h2>
+    <p><%= post.body %></p>
+    <% if user_signed_in? %>
+    <span>written By <%= post.user.name %></span>
+    <% end %>
```
Comment (maelfosso) [inline app/views/posts/index.html.erb:12]: logic must be implemented into the **view helpers**

File: README.md [hunk 6]
```diff
@@ -1,24 +1,75 @@
...
+```
+cd [directory-name]
+```
+6. Bulk install dependencies
+```
+bundle
+```
+7. Set up database
+```
+rake db:migrate
```
Comment (maelfosso) [inline README.md:44]: I think that before `rake db:migrate` there is generally `rails db:create`
Also, be more precise and indicate `bundle install`

**Comment by cliftondavies** [comment]
## Thanks @kobiyoyo 👍 🙏

**Pull request merged**